core/templates/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -432,95 +470,124 @@ angular.module('oppia').directive('questionsList', [
+                categorizedSkills: () => ctrl.getSkillsCategorizedByTopics,
+                allowSkillsFromOtherTopics: () => allowSkillsFromOtherTopics,
+                untriagedSkillSummaries: () => ctrl.getUntriagedSkillSummaries
               },
-              controller: 'QuestionEditorModalController',
-            }).result.then(function(modalObject) {
-              ContextService.resetImageSaveDestination();
-              $location.hash(null);
-              ctrl.editorIsOpen = false;
-              if (modalObject.skillLinkageModificationsArray.length > 0) {
-                EditableQuestionBackendApiService.editQuestionSkillLinks(
-                  questionId, modalObject.skillLinkageModificationsArray,
-                  questionDifficulty).then(
-                  data => {
-                    $timeout(function() {
-                      QuestionsListService.resetPageNumber();
-                      _reInitializeSelectedSkillIds();
-                      ctrl.getQuestionSummariesAsync(
-                        ctrl.selectedSkillIds, true, true
-                      );
-                      ctrl.saveAndPublishQuestion(modalObject.commitMessage);
-                    }, 500);
-                  });
-              } else {
-                ContextService.resetImageSaveDestination();
-                ctrl.saveAndPublishQuestion(modalObject.commitMessage);
+              controller: 'SelectSkillModalController',
+              windowClass: 'skill-select-modal',
+              size: 'xl'
+            }).result.then(function(summary) {
+              for (var idx in ctrl.associatedSkillSummaries) {
+                if (
+                  ctrl.associatedSkillSummaries[idx].getId() ===
+                    summary.id) {
+                  AlertsService.addInfoMessage(
+                    'Skill already linked to question');
+                  return;
+                }
               }
+
+              ctrl.associatedSkillSummaries.push(
+                ShortSkillSummaryObjectFactory.create(
+                  summary.id, summary.description));
+              ctrl.skillLinkageModificationsArray = [];
+              ctrl.skillLinkageModificationsArray.push({
+                id: summary.id,
+                task: 'add'
+              });
             }, function() {
-              ContextService.resetImageSaveDestination();
-              ctrl.editorIsOpen = false;
-              $location.hash(null);
+              // Note to developers:
+              // This callback is triggered when the Cancel button is
+              // clicked. No further action is needed.
             });
           };
 
+          ctrl.updateSkillLinkage = function(commitMsg) {
+            EditableQuestionBackendApiService.editQuestionSkillLinks(
+              ctrl.questionId, ctrl.skillLinkageModificationsArray,
+              ctrl.difficulty).then(
+              data => {
+                $timeout(function() {
```

Comment:
Why do we need a $timeout?

Reply:
I followed the convention that was being followed earlier. Its because we make another call after the first one is completed.

Reply:
Hm, I'm not sure I understand. Can you please tell me what calls you are referring to as "another call" and "first one"?

Reply:
The first one is the ```EditableQuestionBackendApiService.editQuestionSkillLinks``` and then the ```                  EditableQuestionBackendApiService.updateQuestion```.

Reply:
`EditableQuestionBackendApiService.updateQuestion` would execute only after `EditableQuestionBackendApiService.editQuestionSkillLinks` would execute because it is in the success callback, right?

Can you please investigate why this was needed and add a comment accordingly. You can talk to the original author if you are not sure.

Reply:
Ok noted.